File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        int k =0;
        ListNode node = head;
        while(node.next!=null)
        {
            k++;
            node=node.next;
        }
        n=k-n;
        if(n==-1)
        return head.next;

        node = head;
        while(n>0)
        {
            node=node.next;
            n--;
        }
        node.next=node.next.next;
        return head;
    }
}

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        node = head
        size = 0
        while(node):
            size +=1
            node = node.next
        
        target = size - n + 1
        node = head
        prev = None
        for i in range(1, target):
            prev = node
            node = node.next
        if prev and node.next:
            prev.next = node.next
        elif prev:
            prev.next = None
        elif node.next:
            head = node.next
        else:
            return None
        return head